src/CommunityToolkit.Aspire.Hosting.Supabase/SupabaseBuilderExtensions.cs

```diff
@@ -0,0 +1,244 @@
+/// </summary>
+public static class SupabaseBuilderExtensions
+{
+    private const int SupabaseApiPort = 8000;
+    private const int SupabaseDatabasePort = 5432;
+
+    /// <summary>
+    /// Adds a Supabase core Postgres and HTTP API container to the application model.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the Supabase resource.</param>
+    /// <param name="password">Optional password parameter resource builder.</param>
+    /// <param name="apiPort">Optional host port for the Supabase HTTP API.</param>
+    /// <param name="dbPort">Optional host port for the PostgreSQL database.</param>
+    /// <returns>A <see cref="IResourceBuilder{SupabaseResource}"/> for further configuration.</returns>
+    public static IResourceBuilder<SupabaseResource> AddSupabase(
+        this IDistributedApplicationBuilder builder,
+        string name,
+        IResourceBuilder<ParameterResource>? password = null,
+        int? apiPort = 8080,
+        int? dbPort = null)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentNullException.ThrowIfNull(name);
+
+        ParameterResource passwordParam = password?.Resource ?? ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter(builder, $"{name}-password");
+        SupabaseResource resource = new(name, passwordParam);
+
+        return builder.AddResource(resource)
+            .WithImage(SupabaseContainerImageTags.PostgresImage, SupabaseContainerImageTags.PostgresTag)
+            .WithImageRegistry(SupabaseContainerImageTags.Registry)
+            .WithHttpEndpoint(targetPort: SupabaseApiPort, port: apiPort, name: SupabaseResource.PrimaryEndpointName)
+            .WithEndpoint(targetPort: SupabaseDatabasePort, port: dbPort, name: SupabaseResource.DatabaseEndpointName)
+            .WithEnvironment(context =>
+            {
+                context.EnvironmentVariables["POSTGRES_PASSWORD"] = resource.PasswordParameter;
+                context.EnvironmentVariables["POSTGRES_DB"] = "postgres";
+                context.EnvironmentVariables["POSTGRES_USER"] = "postgres";
+                context.EnvironmentVariables["PGDATA"] = "/var/lib/postgresql/data";
+                context.EnvironmentVariables["PGHOST"] = resource.DatabaseEndpoint.Property(EndpointProperty.Host);
+                context.EnvironmentVariables["PGPORT"] = resource.DatabaseEndpoint.Property(EndpointProperty.Port);
+                context.EnvironmentVariables["PGPASSWORD"] = resource.PasswordParameter;
+                context.EnvironmentVariables["PGDATABASE"] = "postgres";
+            })
+            .WithDataBindMount("./volumes/db/data")
+            .WithBindMount("./volumes/db/migrations", "/docker-entrypoint-initdb.d/migrations")
+            .WithBindMount("./volumes/db/init-scripts", "/docker-entrypoint-initdb.d/init-scripts");
```

Comment:
I think it'd be better to break these out as their own methods, like other integrations, so that people can opt-in on data persistence and the other mount points.